### DISCUSSION TITLE
Component Library？

### DISCUSSION THREAD

**Discussion by oliverzy** [solidjs/solid#349] [Q&A]
React & Vue has excellent open source component library support to work with. Is it possible to port them to solid，or I have to write them from scratch？

And what is the matrurity level of solid，is it safe to use in production？

We are in transition from Vue2 to Vue3， and it seems solid is a great alternative as well

**Comment by ryansolid** [comment]
I know a few people have been attempting to build some component libraries but nothing is there yet I could promote other than Lume, but that's not a typical Component library as it is built for 3D rendering and it is Web Component based which has some tradeoffs.

The port to Solid is doable but it would take work. Like it isn't one for one with either framework. Vue would be closer on the logic side and React on the templating code but they aren't the same. It would take careful consideration in each component.

As for stability, most client side APIs are stable and have been for a while. Although in anticipating the release there are some last minute tweaks we are making to naming before putting the stamp on it. Some people have been using Solid in production for smaller client only projects. Resource/SSR API's are less proven. It's still an area I'm actively working and wouldn't consider server use to be production read yet.

**Comment by oliverzy** [comment]
Ok, understood.
We are using Element UI<https://element-plus.org/>. It's a very popular desktop UI framework in China especially.
Maybe it's better to wait for 1.0 and something like solid-router and better documentation so that we can use it in production.

Maybe I can use solid first in my personal website:)

**Comment by ryansolid** [comment]
Yeah, that seems pragmatic. A couple of the routers are in a decent spot right now. I personally am quite liking https://github.com/rturnq/solid-router, which is up to date and decently featured. As you can imagine documentation will improve, I imagine component libraries will take time, as they have for Svelte or library coming up. Someone needs to make them.

**Comment by oliverzy** [comment]
Right. Even now lots of Vue3 component libraries are in beta quality including Element UI.
Anyway, I'm a long-term Vue user since 1.x. I still remember at that time when I ported my Augular 1.x app to Vue 1.x in one month. I never want to go back.

I have a very good first impression of Solid and I think it has both advantages of Vue/React and without vdom makes it truly outstanding.

I hope I can use Solid ASAP and even port a library like Element UI to solid if I have time.

I think Solid is the next generation UI Frameworks compared to Vue/React/Angular.
But I’ll continue using Vue3 at the time being.

Hopefully this year will be the year of Solid.  And I definitely want to give it a serious try after 1.0 release.
